reflect.go

```diff
@@ -0,0 +1,388 @@
+package main
+
+import (
+	"fmt"
+	"go/types"
+	"path/filepath"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/tools/go/ssa"
+)
+
+var checkedAPIs = make(map[string]bool)
+
+// Record all instances of reflection use, and don't obfuscate types which are used in reflection.
+func (tf *transformer) recordReflection(ssaPkg *ssa.Package) {
+	lenPrevKnownReflectAPIs := len(cachedOutput.KnownReflectAPIs)
+
+	notCheckedAPIs := make(map[string]bool)
+	for _, knownAPI := range maps.Keys(cachedOutput.KnownReflectAPIs) {
+		if !checkedAPIs[knownAPI] {
+			notCheckedAPIs[knownAPI] = true
+		}
+	}
+
+	tf.ignoreReflectedTypes(ssaPkg)
+
+	maps.Copy(checkedAPIs, notCheckedAPIs)
+
+	// if a new reflectAPI is found we need to Re-evaluate all functions which might be using that API
+	if len(cachedOutput.KnownReflectAPIs) > lenPrevKnownReflectAPIs {
+		tf.recordReflection(ssaPkg)
```

Comment:
this makes me nervous, as we could potentially loop forever. presumably we only need to run the func on ssaPkg once or twice - how about a `goto` with a boolean to ensure we don't run more than twice?

Reply:
I'm not worried about looping forever as this only recurses when a new function which uses reflection on its parameters is found. It is then only checked if that new function is used in a call which I find reasonable. Packages routinely require more than two passes (3-4) is common.

Reply:
ack, makes sense. I guess we can always reconsider if we encounter endless loops in practice.